spec/support/fake_ansible_repo.rb

```diff
@@ -0,0 +1,357 @@
+      #
+      #   - Array:  Collection of directory contents (either Strings or Hashes)
+      #   - String: A single filename
+      #   - Hash:   A collection (usually one) directory (key) with entries (value)
+      #
+      # The format can be delivered in two forms.  Either pure yaml:
+      #
+      #     - path/:
+      #       - to/:
+      #         - dir:
+      #           - foo
+      #           - bar
+      #           - baz
+      #     - other/:
+      #       - dir/:
+      #
+      # Or an plain text version that without some of the extra yaml chars added:
+      #
+      #     path/
+      #       to/
+      #         dir/
+      #           foo
+      #           bar
+      #           baz
+      #     other/
+      #       dir/
+      #
+      #
+      # Which is a cheeky trick with some regular expressions to allow for a
+      # more human readable format.  Both still go through yaml, and the extra
+      # formatting will be added only when missing.
+      #
+      # To use the second form, however, directories require a trailing slash,
+      # where they are optional in the yaml form (the `:` takes care of that).
+      #
+      def parse_file_struct(file_struct)
+        yaml_data = file_struct.gsub(/^(\s*)([a-zA-Z0-9\.])/, '\1- \2')
+                               .gsub(/\/$/, '/:')
+        YAML.load(yaml_data)
+      end
+
+      # Builds a (nested) directory structure at the given path and file
+      # structure definition.
+      #
+      # This method is called recursively, so `repo_relpath` and `file_struct`
+      # might be in a nested partial state depending how deep the call of this
+      # method has made it.
+      #
+      def build_files_at(repo_rel_path, file_struct)
+        file_struct.each do |entry|
+          case entry
+          when Array  then build_files_at(repo_rel_path, entry)
+          when Hash   then build_directory(repo_rel_path, entry)
+          when String then build_file(repo_rel_path, entry)
+          end
+        end
+      end
+
+      # Generates a single directory (hash key)
+      #
+      # If there is entries (value), it defers back to `.build_files_at`,
+      # otherwise if it is `nil` it represents a empty dir.
+      #
+      def build_directory(repo_rel_path, file_struct)
+        file_struct.each do |dir, entries|
+          new_dir = repo_rel_path.join(dir.sub(/\/$/, ''))
+
+          FileUtils.mkdir_p(new_dir)
+          build_files_at(new_dir, entries) if entries
+        end
+      end
+
+      VAR_FILE_FNMATCHES = [
+        "**/extra_vars.{yml,yaml}",
+        "**/roles/*/vars/*.{yml,yaml}",
+        "**/roles/*/defaults/*.{yml,yaml}"
+      ]
+
+      # Gerates a single file
+      #
+      # If it is a reserved file name/path, then the contents are generated,
+      # otherwise it will be a empty file.
+      #
+      def build_file(repo_rel_path, entry)
+        full_path = repo_rel_path.join(entry)
+
+        File.open(full_path, "w") do |new_file|
```

Comment:
Minor but I feel this is cleaner as 

```ruby
content =
  case
  when filepath_match?(full_path, "**/requirements.{yml,yaml}")
    REQUIREMENTS
  when filepath_match?(full_path, *VAR_FILE_FNMATCHES)
    VAR_DATA
  when filepath_match?(full_path, "**/roles/*/meta/main.{yml,yaml}")
    META_DATA
  when filepath_match?(full_path, "**/*.{yml,yaml}")
    dummy_playbook_data_for(full_path.basename)
  end

File.write(full_path, content)
```

Reply:
I hate "hanging assignments" (not sure if that is official terminology) like you have done above, but I can see what I can do about getting closer to your suggestion.